engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -2386,6 +2399,9 @@ public void deploy(Set<PTContainer> releaseContainers, Collection<PTOperator> un
   @Override
   public void recordEventAsync(StramEvent ev)
   {
+    if (apexPluginManager != null) {
```

Comment:
Can the plugin manager listen to the event bus, get the event from the bus and use it? It might be able to get access to other types of events as well by listening to the bus.